hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/IncrementalBlockReportManager.java

```diff
@@ -169,11 +169,17 @@ private synchronized StorageReceivedDeletedBlocks[] generateIBRs() {
         : pendingIBRs.entrySet()) {
       final PerStorageIBR perStorage = entry.getValue();
 
-        // Send newly-received and deleted blockids to namenode
+      // Send newly-received and deleted blockids to namenode
       final ReceivedDeletedBlockInfo[] rdbi = perStorage.removeAll();
-      if (rdbi != null) {
-        reports.add(new StorageReceivedDeletedBlocks(entry.getKey(), rdbi));
+      if (rdbi == null) {
```

Comment:
```
if (rdbi != null) {
  // Null storage, should not happen
  if (entry.getKey() == null) {
    dnMetrics.incrNullStorageBlockReports();
    continue;
  }
  reports.add(new StorageReceivedDeletedBlocks(entry.getKey(), rdbi));
}
```

Reply:
I intentionally put it like this to reduce nesting and make the logic a bit easier to read